.travis.yml

```diff
@@ -0,0 +1,34 @@
+# use java support
+language: java
+jdk:
+  - openjdk11
+
+# run in container
+sudo: false
+
+# use linux as operating system
+os:
+  - linux
+
+# run maven build
+script:
+  - ./gradlew clean build
+
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+
+# Deploy snapshots on every commit made to master branch
+# Deploy releases on every tag push
+deploy:
+  - provider: script
+    script: ./gradlew publish -PrepositoryUrl=$url/public-snapshots/ -PrepositoryUsername=$username -PrepositoryPassword=$password
```

Comment:
Shouldn't we prepend the variables `$url`, `$username` and `$password` with `repository`, so they become: `$repositoryUrl`, `$repositoryUsername` and `$repositoryPassword` respectively?

Reply:
why would we want do that?

Reply:
These variables are like global variables, so I would name them so that it's clear what they stand for. `url` could be anything, while `repositoryUrl` is more specific. In that retrospect, `mavenPublish` might be even a better prefix.

Reply:
Ok, makes sense. I will append mavenPublish to the variable names.